### ISSUE TITLE
ExceptionInInitializerError on Spigot 1.14.4

### ISSUE THREAD

**Issue by SlimeDog** [flori-schwa/VarLightOld#6]
Spigot 1.14.4
VarLight 1.3.5.3

On server start-up
```
[17:07:56] [Server thread/ERROR]: Error occurred while enabling VarLight v1.3.5.3 (Is it up to date?)
java.lang.ExceptionInInitializerError: null
        at xyz.upperlevel.spigot.book.BookUtil$BookBuilder.pages(BookUtil.java:136) ~[?:?]
        at me.florian.varlight.command.VarLightCommand.<init>(VarLightCommand.java:54) ~[?:?]
        at me.florian.varlight.VarLightPlugin.onEnable(VarLightPlugin.java:179) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:352) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:417) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugin(CraftServer.java:461) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugins(CraftServer.java:375) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:449) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at net.minecraft.server.v1_14_R1.DedicatedServer.init(DedicatedServer.java:258) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:776) [spigot-1.14.4.jar-2019-07-19-1706:git-Spigot-9de398a-9c887d4]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalStateException: Cannot initiate reflections for v1_14_R1
        at xyz.upperlevel.spigot.book.NmsBookHelper.<clinit>(NmsBookHelper.java:104) ~[?:?]
        ... 12 more
Caused by: java.lang.NoSuchMethodException: net.minecraft.server.v1_14_R1.EntityPlayer.a(net.minecraft.server.v1_14_R1.ItemStack, net.minecraft.server.v1_14_R1.EnumHand)
        at java.lang.Class.getMethod(Class.java:2109) ~[?:?]
        at xyz.upperlevel.spigot.book.NmsBookHelper.<clinit>(NmsBookHelper.java:81) ~[?:?]
        ... 12 more
```

**Comment by flori-schwa** [comment]
BookAPI seems to be broken in 1.14.4

**Comment by SlimeDog** [comment]
Should we assume that you have posted an appropriate ticket?

**Comment by flori-schwa** [comment]
https://github.com/upperlevel/book-api has been inactive since 2017 and my pull request upperlevel/book-api#5 from May is still open. I will write a patch for 1.14.4 and submit another PR (which will probably also be ignored) and include the patch in a future release, since BookAPI is shaded into the final jar.

**Comment by SlimeDog** [comment]
Thanks.

**Comment by flori-schwa** [comment]
Just discovered another nasty thing about 1.14.4:

![](https://i.imgur.com/LNZzheo.png)

Why don't they change the package version? Why is it still "__v1_14_R1__" for __1.14.4__?

I will probably have to rewrite parts of the plugin if I don't want to keep on making "hacky" fixes for stupid stuff like this

**Comment by flori-schwa** [comment]
__This__ specific issue was fixed in upperlevel/book-api#7 and the fixed book-api will be included in future releases of VarLight. The plugin however is still broken for 1.14.4 throwing AbstractMethodErrors.

**Comment by SnowyCoder** [comment]
@flori-schwa hello, I'm part of the upperlevel team (developers of the book api). We have been inactive for a while because of school and now we're on holiday but we haven't abandoned it. 
Thanks for the support! I'll fix it as soon as I can.
